workspaces/services/guacamole/terraform/web_app.tf

```diff
@@ -0,0 +1,85 @@
+resource "azurerm_app_service_plan" "guacamole" {
+  name                = "asp-inf-${local.service_resource_name_suffix}"
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+  kind                = "Linux"
+  reserved            = "true"
+
+  sku {
+    tier = "PremiumV2"
+    size = "P1v2"
+  }
+}
+
+resource "azurerm_app_service" "guacamole" {
+  name                = "app-gua-${local.service_resource_name_suffix}"
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+  app_service_plan_id = azurerm_app_service_plan.guacamole.id
+  https_only          = true
+
+  site_config {
+    linux_fx_version = "DOCKER|${var.acr_name}.azurecr.io/guac-server:${var.guacamole_image_tag}"
+    always_on        = true
+    http2_enabled    = true
+  }
+
+  app_settings = {
+    "WEBSITE_VNET_ROUTE_ALL"         = "1"
+    "WEBSITE_DNS_SERVER"             = "168.63.129.16"
+    "SCM_DO_BUILD_DURING_DEPLOYMENT" = "True"
+
+    "RESOURCE_GROUP"    = data.azurerm_resource_group.ws.name
+    "SUBSCRIPTION_ID"   = data.azurerm_client_config.current.subscription_id
+    "TENANT_ID"         = data.azurerm_client_config.current.tenant_id
+
+    WEBSITES_PORT       = "8080"
+  }
+  
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+# Use managed identity to login to ACR
+# https://github.com/Azure/app-service-linux-docs/blob/master/HowTo/use_system-assigned_managed_identities.md
+resource "null_resource" "az_login" {
+  provisioner "local-exec" {
+    # command = "az login --identity -u '${data.azurerm_client_config.current.client_id}'"
+    command = "az login --service-principal --username ${var.arm_client_id} --password ${var.arm_client_secret} --tenant ${var.arm_tenant_id} && az resource update --ids ${azurerm_app_service.guacamole.id}/config/web --set properties.acrUseManagedIdentityCreds=True -o none"
```

Comment:
this should use managed identity - again the local vs remote run is a challenge.

Reply:
Perhaps the command can check if the arm_client_id is set and then use it. otherwise to go with managed identity.

Reply:
if sp provided -> use it.
if sp not provided and has managed identity -> use it

Reply:
@eladiw  ARM CLIENT ID will be set in both scenarios - best to use `use_arm_msi`

Reply:
fixed